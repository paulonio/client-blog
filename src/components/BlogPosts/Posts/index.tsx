import React, { FC } from 'react';
import { useTranslation } from 'next-i18next';
import Link from 'next/link';

import Post from '@/components/Post';

import { PostType } from '@/types/types';

import styles from './styled.module.scss';

interface PostsProps {
  posts: PostType[];
}

const Posts: FC<PostsProps> = ({ posts }) => {
  const { t } = useTranslation('blog');

  if (!posts.length) {
    return (
      <div className={styles.posts__wrapper}>
        <h2 className={styles.message}>{t('no-result')}</h2>
      </div>
    );
  }

  return (
    <div className={styles.posts__wrapper}>
      {posts.map((post) => (
        <Link key={post.id} href={`/blog/${post.id}`}>
          <Post key={`${post.id}`} post={post} />
        </Link>
      ))}
    </div>
  );
};

export default Posts;
