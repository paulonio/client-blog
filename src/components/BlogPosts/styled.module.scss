@import '../../styles/sass/constants';

.posts {


  & h1 {
    margin: 64px 0;
    height: 96px;
    border-bottom: solid 1px $color-dark-s;
    color: $color-black;
  }

  &__wrapper {
    display: flex;
    flex-direction: column;
    row-gap: 64px;
  }
}

.buttons {
  margin: 64px 0;
  display: flex;
  justify-content: center;
  gap: 17px;
}

.button {
  font-size: 28px;
  line-height: 40px;
  font-weight: 700;
  letter-spacing: -1px;
  color: $color-black;
  transition: color 0.3s;
  background-color: transparent;
  border: none;


  &:enabled:hover {
    color: $color-secondary;
    cursor: pointer;
  }

  &:disabled {
    color: $color-dark-s;
  }
}