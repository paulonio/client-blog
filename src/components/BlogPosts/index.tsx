'use client';

import React from 'react';
import { useTranslation } from 'next-i18next';

import Posts from './Posts';

import { usePagination } from '@/hooks/usePagination';

import { sen } from '@/styles/fonts';

import styles from './styled.module.scss';

const BlogPosts = () => {
  const { posts, isPrevDisabled, isNextDisabled, handleNextClick, handlePrevClick } =
    usePagination();
  const { t } = useTranslation('blog');

  return (
    <section className={styles.posts}>
      <h1 className={`${sen.className} ${styles.posts__title}`} data-testid="blog-posts-title">
        {t('posts')}
      </h1>
      <Posts posts={posts} />
      <div className={styles.buttons}>
        <button
          type="button"
          disabled={isPrevDisabled}
          className={styles.button}
          onClick={handlePrevClick}
          data-testid="blog-posts-prev"
        >
          {'<'} {t('prev')}{' '}
        </button>
        <button
          type="button"
          disabled={isNextDisabled}
          className={styles.button}
          onClick={handleNextClick}
          data-testid="blog-posts-next"
        >
          {t('next')} {'>'}
        </button>
      </div>
    </section>
  );
};

export default BlogPosts;
