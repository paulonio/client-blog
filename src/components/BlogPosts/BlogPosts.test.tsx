import React from 'react';
import { fireEvent, render, screen } from '@testing-library/react';
import BlogPosts from '.';

describe('Blog posts', () => {
  it('should render blog posts component', () => {
    const { container } = render(<BlogPosts />);

    expect(container).toMatchSnapshot();
  });

  it('should handle buttons click', () => {
    const { container } = render(<BlogPosts />);

    const prev = screen.getByTestId('blog-posts-prev');
    const next = screen.getByTestId('blog-posts-next');
    expect(prev).toBeDisabled();
    fireEvent.click(next);
    expect(prev).not.toBeDisabled();

    expect(container).toMatchSnapshot();
  });
});
