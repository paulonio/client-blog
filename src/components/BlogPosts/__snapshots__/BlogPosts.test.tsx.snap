// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Blog posts should handle buttons click 1`] = `
<div>
  <section
    class="posts"
  >
    <h1
      class="className posts__title"
      data-testid="blog-posts-title"
    >
      posts
    </h1>
    <div
      class="posts__wrapper"
    >
      <a
        href="/blog/6"
      >
        <div
          class="post"
        >
          <div
            class="image"
          >
            <img
              alt="Business post"
              src="/assets/images/figma-design.jpg"
            />
          </div>
          <div
            class="content"
          >
            <h5>
              business
            </h5>
            <h2>
              lorem-ipsum
            </h2>
            <p
              class="body-s"
            >
              duis-aute
            </p>
          </div>
        </div>
      </a>
      <a
        href="/blog/7"
      >
        <div
          class="post"
        >
          <div
            class="image"
          >
            <img
              alt="Business post"
              src="/assets/images/featured-post.jpg"
            />
          </div>
          <div
            class="content"
          >
            <h5>
              business
            </h5>
            <h2>
              lorem-ipsum
            </h2>
            <p
              class="body-s"
            >
              duis-aute
            </p>
          </div>
        </div>
      </a>
      <a
        href="/blog/8"
      >
        <div
          class="post"
        >
          <div
            class="image"
          >
            <img
              alt="Technology post"
              src="/assets/images/featured-post.jpg"
            />
          </div>
          <div
            class="content"
          >
            <h5>
              technology
            </h5>
            <h2>
              lorem-ipsum
            </h2>
            <p
              class="body-s"
            >
              duis-aute
            </p>
          </div>
        </div>
      </a>
      <a
        href="/blog/9"
      >
        <div
          class="post"
        >
          <div
            class="image"
          >
            <img
              alt="Business post"
              src="/assets/images/featured-post.jpg"
            />
          </div>
          <div
            class="content"
          >
            <h5>
              business
            </h5>
            <h2>
              lorem-ipsum
            </h2>
            <p
              class="body-s"
            >
              duis-aute
            </p>
          </div>
        </div>
      </a>
      <a
        href="/blog/10"
      >
        <div
          class="post"
        >
          <div
            class="image"
          >
            <img
              alt="Startup post"
              src="/assets/images/blog-featured.jpg"
            />
          </div>
          <div
            class="content"
          >
            <h5>
              startup
            </h5>
            <h2>
              step-by-step-guide
            </h2>
            <p
              class="body-s"
            >
              duis-aute
            </p>
          </div>
        </div>
      </a>
    </div>
    <div
      class="buttons"
    >
      <button
        class="button"
        data-testid="blog-posts-prev"
        type="button"
      >
        &lt;
         
        prev
         
      </button>
      <button
        class="button"
        data-testid="blog-posts-next"
        type="button"
      >
        next
         
        &gt;
      </button>
    </div>
  </section>
</div>
`;

exports[`Blog posts should render blog posts component 1`] = `
<div>
  <section
    class="posts"
  >
    <h1
      class="className posts__title"
      data-testid="blog-posts-title"
    >
      posts
    </h1>
    <div
      class="posts__wrapper"
    >
      <a
        href="/blog/1"
      >
        <div
          class="post"
        >
          <div
            class="image"
          >
            <img
              alt="Business post"
              src="/assets/images/featured-post.jpg"
            />
          </div>
          <div
            class="content"
          >
            <h5>
              business
            </h5>
            <h2>
              lorem-ipsum
            </h2>
            <p
              class="body-s"
            >
              duis-aute
            </p>
          </div>
        </div>
      </a>
      <a
        href="/blog/2"
      >
        <div
          class="post"
        >
          <div
            class="image"
          >
            <img
              alt="Startup post"
              src="/assets/images/blog-featured.jpg"
            />
          </div>
          <div
            class="content"
          >
            <h5>
              startup
            </h5>
            <h2>
              step-by-step-guide
            </h2>
            <p
              class="body-s"
            >
              duis-aute
            </p>
          </div>
        </div>
      </a>
      <a
        href="/blog/3"
      >
        <div
          class="post"
        >
          <div
            class="image"
          >
            <img
              alt="Startup post"
              src="/assets/images/design-post.jpg"
            />
          </div>
          <div
            class="content"
          >
            <h5>
              startup
            </h5>
            <h2>
              lorem-ipsum
            </h2>
            <p
              class="body-s"
            >
              duis-aute
            </p>
          </div>
        </div>
      </a>
      <a
        href="/blog/4"
      >
        <div
          class="post"
        >
          <div
            class="image"
          >
            <img
              alt="Technology post"
              src="/assets/images/build-rapport.jpg"
            />
          </div>
          <div
            class="content"
          >
            <h5>
              technology
            </h5>
            <h2>
              lorem-ipsum
            </h2>
            <p
              class="body-s"
            >
              duis-aute
            </p>
          </div>
        </div>
      </a>
      <a
        href="/blog/5"
      >
        <div
          class="post"
        >
          <div
            class="image"
          >
            <img
              alt="Economy post"
              src="/assets/images/design-post.jpg"
            />
          </div>
          <div
            class="content"
          >
            <h5>
              economy
            </h5>
            <h2>
              lorem-ipsum
            </h2>
            <p
              class="body-s"
            >
              duis-aute
            </p>
          </div>
        </div>
      </a>
    </div>
    <div
      class="buttons"
    >
      <button
        class="button"
        data-testid="blog-posts-prev"
        disabled=""
        type="button"
      >
        &lt;
         
        prev
         
      </button>
      <button
        class="button"
        data-testid="blog-posts-next"
        type="button"
      >
        next
         
        &gt;
      </button>
    </div>
  </section>
</div>
`;
